tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -175,14 +175,30 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
             return Extract(serializedSpanContext, default(ReadOnlyDictionaryGetter));
         }
 
-        public IEnumerable<KeyValuePair<string, string?>> ExtractHeaderTags<T>(T headers, IEnumerable<KeyValuePair<string, string?>> headerToTagMap, string defaultTagPrefix)
-            where T : IHeadersCollection
+        public void ExtractHeaderTags<THeaders>(ISpan span, THeaders headers, IEnumerable<KeyValuePair<string, string?>> headerToTagMap, string defaultTagPrefix)
+            where THeaders : IHeadersCollection
         {
-            return ExtractHeaderTags(headers, headerToTagMap, defaultTagPrefix, string.Empty);
+            var processor = new SpanTagHeaderTagProcessor(span);
+            ExtractHeaderTags(ref processor, headers, headerToTagMap, defaultTagPrefix, string.Empty);
         }
 
-        public IEnumerable<KeyValuePair<string, string?>> ExtractHeaderTags<T>(T headers, IEnumerable<KeyValuePair<string, string?>> headerToTagMap, string defaultTagPrefix, string useragent)
-            where T : IHeadersCollection
+        public void ExtractHeaderTags<THeaders>(ISpan span, THeaders headers, IEnumerable<KeyValuePair<string, string?>> headerToTagMap, string defaultTagPrefix, string useragent)
+            where THeaders : IHeadersCollection
+        {
+            var processor = new SpanTagHeaderTagProcessor(span);
+            ExtractHeaderTags(ref processor, headers, headerToTagMap, defaultTagPrefix, useragent);
+        }
+
+        public void ExtractHeaderTags<THeaders, TProcessor>(ref TProcessor processor, THeaders headers, IEnumerable<KeyValuePair<string, string?>> headerToTagMap, string defaultTagPrefix)
+            where THeaders : IHeadersCollection
+            where TProcessor : struct, IHeaderTagProcessor
+        {
+            ExtractHeaderTags(ref processor, headers, headerToTagMap, defaultTagPrefix, string.Empty);
+        }
+
+        public void ExtractHeaderTags<THeaders, TProcessor>(ref TProcessor processor, THeaders headers, IEnumerable<KeyValuePair<string, string?>> headerToTagMap, string defaultTagPrefix, string useragent)
```

Comment:
I'm having this warning when using `in` instead of `ref`

<img width="787" alt="image" src="https://github.com/DataDog/dd-trace-dotnet/assets/69803/8718d322-90cd-4d3e-b52b-2c48a82ee52c">

Reply:
I think you can remove this warning by making the `ProcessTag()` method `readonly`.